TrafficCapture/trafficReplayer/src/main/java/org/opensearch/migrations/replay/datahandlers/http/NettyJsonBodySerializeHandler.java

```diff
@@ -22,40 +23,46 @@ public class NettyJsonBodySerializeHandler extends ChannelInboundHandlerAdapter
-            } else if (payload.containsKey(JsonKeysForHttpMessage.INLINED_NDJSON_BODIES_DOCUMENT_KEY)) {
-                serializePayloadList(ctx,
-                    (List) payload.get(JsonKeysForHttpMessage.INLINED_NDJSON_BODIES_DOCUMENT_KEY),
-                    !payload.containsKey(JsonKeysForHttpMessage.INLINED_BINARY_BODY_DOCUMENT_KEY) &&
-                        !payload.containsKey(JsonKeysForHttpMessage.INLINED_TEXT_BODY_DOCUMENT_KEY));
-            }
-            if (payload.containsKey(JsonKeysForHttpMessage.INLINED_BINARY_BODY_DOCUMENT_KEY)) {
-                var rawBody = (ByteBuf) payload.get(JsonKeysForHttpMessage.INLINED_BINARY_BODY_DOCUMENT_KEY);
-                if (rawBody.readableBytes() > 0) {
-                    ctx.fireChannelRead(new DefaultHttpContent(rawBody));
-                } else {
-                    ReferenceCountUtil.release(rawBody);
-                }
-            } else if (payload.containsKey(JsonKeysForHttpMessage.INLINED_TEXT_BODY_DOCUMENT_KEY)) {
-                var bodyString = (String) payload.get(JsonKeysForHttpMessage.INLINED_TEXT_BODY_DOCUMENT_KEY);
-                ByteBuf body = ctx.alloc().buffer();
-                body.writeCharSequence(bodyString, StandardCharsets.UTF_8);
-                if (body.readableBytes() > 0) {
-                    ctx.fireChannelRead(new DefaultHttpContent(body));
-                } else {
-                    ReferenceCountUtil.release(body);
-                }
-            }
-            ctx.fireChannelRead(LastHttpContent.EMPTY_LAST_CONTENT);
-        } else {
+        if (!(msg instanceof HttpJsonRequestWithFaultingPayload)) {
             super.channelRead(ctx, msg);
         }
+        ctx.fireChannelRead(msg);
+
+        var jsonMessage = (HttpJsonRequestWithFaultingPayload) msg;
+        var payload = jsonMessage.payload();
+        jsonMessage.setPayloadFaultMap(null);
+
+        processInlineDocuments(ctx, payload);
+    }
+
+    @SuppressWarnings("unchecked")
+    private void processInlineDocuments(ChannelHandlerContext ctx, Map<String, Object> payload) throws Exception {
+        if (payload.containsKey(JsonKeysForHttpMessage.INLINED_JSON_BODY_DOCUMENT_KEY)) {
+            serializePayload(ctx, payload.get(JsonKeysForHttpMessage.INLINED_JSON_BODY_DOCUMENT_KEY));
+        } else if (payload.containsKey(JsonKeysForHttpMessage.INLINED_NDJSON_BODIES_DOCUMENT_KEY)) {
+            var shouldAddLastNewline = !payload.containsKey(JsonKeysForHttpMessage.INLINED_BINARY_BODY_DOCUMENT_KEY) &&
+                !payload.containsKey(JsonKeysForHttpMessage.INLINED_TEXT_BODY_DOCUMENT_KEY);
+            serializePayloadList(ctx,
+                (List<Object>) payload.get(JsonKeysForHttpMessage.INLINED_NDJSON_BODIES_DOCUMENT_KEY),
+                shouldAddLastNewline);
+        } else if (payload.containsKey(JsonKeysForHttpMessage.INLINED_BINARY_BODY_DOCUMENT_KEY)) {
```

Comment:
Could I get a double check here? Previously the organization of these if statements was 2 separate groups with json/ndjson checked and then a separate checks for binary/text.  That doesn't seem correct, it seems like a document should only include one of json/ndjson/binary/text.

See @38 on the original file for this behavoir

Reply:
Thanks! I've corrected this, just pushed an update.